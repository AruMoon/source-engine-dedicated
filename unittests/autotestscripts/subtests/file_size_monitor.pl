#!perl

use File::Find;

# customize here
print "Running file size monitor\n";

LogDirectorySize("PC shader size", "../../../game/hl2/shaders","\.vcs","\.360\.vcs");
LogDirectorySize("PC Game Bin DLL size", "../../../game/bin/","\.dll","_360\.dll");
LogDirectorySize("360 shader size", "../../../game/hl2/shaders","\.360\.vcs");
LogDirectorySize("360 Game Bin DLL size", "../../../game/bin/","_360\.dll");
LogDirectorySize("tf texture size","../../../game/tf/materials/","\.vtf");







sub LogDirectorySize
  {
	my ($label, $basedir, $filepattern, $excludepattern ) = @_;
	undef @FileList;
	find(\&ProcessFile, $basedir);
	my $total_size = 0;
	foreach $_ (@FileList)
	  {
		next if ( length($excludepattern) && ( /$excludepattern/i ) );
		if (/$filepattern/i)
		{
		  $total_size += (-s $_ );
		}
	  }
	print "$label := $total_size\n";
  }

sub ProcessFile
  {
	push @FileList, $File::Find::name;
  }
