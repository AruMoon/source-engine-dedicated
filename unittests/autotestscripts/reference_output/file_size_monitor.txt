Running file size monitor
PC shader size := 291934326
PC Game Bin DLL size := 108442741
360 shader size := 0
360 Game Bin DLL size := 0
tf texture size := 1396651036
